src/modules/windowwalker/app/Window Walker/Components/Window.cs

```diff
@@ -168,6 +168,28 @@ public bool Visible
             }
         }
 
+        /// <summary>
+        /// Determines whether the specified window handle identifies an existing window.
+        /// </summary>
+        public bool IsWindow
+        {
+            get
+            {
+                return InteropAndHelpers.IsWindow(Hwnd);
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether is the window cloaked. To detect UWP apps in background or win32 apps running in another virtual desktop
+        /// </summary>
+        public bool IsWindowCloaked()

```

Comment:
Does this mean that you can't distinguish between UWP apps in background vs win32 apps running in another Virtual desktop?

Reply:
That's somewhat correct, `DwmGetWindowAttribute` testing `DWMWA_CLOAKED` flag will return `DWM_CLOAKED_SHELL` (**0x02**) value if cloaked, otherwise will return **0** if its not hidden, there are other values too.

Most UWP apps also have its classname set to `ApplicationFrameWindow`, there are similar windows which also ignores `IsWindowVisible` as UWP apps and those have its classname set to `Windows.UI.Core.CoreWindow`, that's why it is excluded.

So in order to find out if a cloaked app is whether a UWP or a Win32 app, after all those comparisons, is just a matter of also comparing their classnames, i.e. a "ApplicationFrameWindow" classname for UWP, not really a nice way though.

Here is the modified part, which also includes (commented out) logic to hide all cloaked apps, or just UWP cloaked apps. If it is planned to configure which to show in the future, playing with them will do the work. 
```cs
if ((newWindow.IsWindow && newWindow.Visible && InteropAndHelpers.GetWindow(hwnd, owner) != null &&
                (((windowLong & toolWindow) != toolWindow) || ((windowLong & appWindow) == appWindow)) &&
                InteropAndHelpers.GetProp(hwnd, "ITaskList_Deleted") == IntPtr.Zero &&
                newWindow.ClassName != "Windows.UI.Core.CoreWindow" /*&& !newWindow.IsWindowCloaked() //hides all cloaked apps*/)
                )
            {
                /*if (newWindow.IsWindowCloaked() && newWindow.ClassName == "ApplicationFrameWindow") 
                {
                    // hides cloaked UWP apps only
                }
                else*/
                    windows.Add(newWindow);
            }
```

Reply:
BTW: virtual desktop exclusion pushed https://github.com/microsoft/PowerToys/pull/1956/commits/7bc5863e98550d1ec0404fdd988a986035528b62

Reply:
One thing I couldn't find a way to detect is fullscreen UWP apps (after entering with Win+Shift+Enter), they seem to work differently.

Reply:
Thanks for the explanation above, I really appreciate you taking the time to explain!

As for full screen UWP apps, I would much prefer that we have a bug in that corner case than not being able to search across virtual desktops.